spread/client.go

```diff
@@ -281,26 +281,32 @@ func (c *Client) run(script string, dir string, env *Environment, mode outputMod
 		rebootKey = rerr.Key
 		output = append(output, '\n')
 
-		timedout := time.After(c.killTimeout)
-		err := c.Run(fmt.Sprintf("reboot &\nsleep %.0f", c.killTimeout.Seconds()), "", nil)
+		uptime, err := c.getUptime()
 		if err != nil {
-			err = c.Run("echo should-have-disconnected", "", nil)
+			return nil, fmt.Errorf("failed to get uptime before reboot, error: %v", err)
 		}
-		if err == nil {
-			select {
-			case <-timedout:
-				return nil, fmt.Errorf("kill-timeout reached on %s while waiting for reboot", c.job)
-			default:
-			}
-			return nil, fmt.Errorf("reboot request on %s failed", c.job)
-		}
-		if err := c.dialOnReboot(); err != nil {
+		c.Run(fmt.Sprintf("reboot"), "", nil)
+		if err := c.dialOnReboot(uptime); err != nil {
 			return nil, err
 		}
 	}
 	panic("unreachable")
 }
 
+func (c *Client) getUptime() (time.Time, error) {
+	uptime, err := c.Output("date -u --iso-8601=seconds -d \"`cut -f1 -d. /proc/uptime` seconds ago\"", "", nil)
+	if err != nil {
+		return time.Now(), err
```

Comment:
`time.Time{}` is the zero value of `time.Time`.

Also, let's please return an actual error message, both here and below, if we cannot get the time:
```
return time.Time{}, fmt.Errorf("cannot obtain the remote system uptime: %v", err)
```